connection_configuration.go

```diff
@@ -0,0 +1,332 @@
+	case "authenticator":
+		v, parsingErr = parseString(value)
+		if err := checkParsingError(parsingErr, key, value); err != nil {
+			return err
+		}
+		parsingErr = determineAuthenticatorType(cfg, v)
+	case "insecuremode":
+		cfg.InsecureMode, parsingErr = parseBool(value)
+	case "ocspfailopen":
+		var vv ConfigBool
+		vv, parsingErr = parseConfigBool(value)
+		if err := checkParsingError(parsingErr, key, value); err != nil {
+			return err
+		}
+		cfg.OCSPFailOpen = OCSPFailOpenMode(vv)
+	case "token":
+		cfg.Token, parsingErr = parseString(value)
+	case "privatekey":
+		v, parsingErr = parseString(value)
+		if err := checkParsingError(parsingErr, key, value); err != nil {
+			return err
+		}
+		block, decodeErr := base64.URLEncoding.DecodeString(v)
+		if decodeErr != nil {
+			return &SnowflakeError{
+				Number:  ErrCodePrivateKeyParseError,
+				Message: "Base64 decode failed",
+			}
+		}
+		cfg.PrivateKey, parsingErr = parsePKCS8PrivateKey(block)
+	case "validatedefaultparameters":
+		cfg.ValidateDefaultParameters, parsingErr = parseConfigBool(value)
+	case "clientrequestmfatoken":
+		cfg.ClientRequestMfaToken, parsingErr = parseConfigBool(value)
+	case "clientstoretemporarycredential":
+		cfg.ClientStoreTemporaryCredential, parsingErr = parseConfigBool(value)
+	case "tracing":
+		cfg.Tracing, parsingErr = parseString(value)
+	case "tmpdirpath":
+		cfg.TmpDirPath, parsingErr = parseString(value)
+	case "disablequerycontextcache":
+		cfg.DisableQueryContextCache, parsingErr = parseBool(value)
+	case "includeretryreason":
+		cfg.IncludeRetryReason, parsingErr = parseConfigBool(value)
+	case "clientconfigfile":
+		cfg.ClientConfigFile, parsingErr = parseString(value)
+	case "disableconsolelogin":
+		cfg.DisableConsoleLogin, parsingErr = parseConfigBool(value)
+	case "disablesamlurlcheck":
+		cfg.DisableSamlURLCheck, parsingErr = parseConfigBool(value)
+	case "token_file_path":
+		tokenPath, parsingErr = parseString(value)
+		if err := checkParsingError(parsingErr, key, value); err != nil {
+			return err
+		}
+		v, err := readToken(tokenPath)
+		if err != nil {
+			return err
+		}
+		cfg.Token = v
+	default:
+		param, parsingErr := parseString(value)
+		if err := checkParsingError(parsingErr, key, value); err != nil {
+			return err
+		}
+		cfg.Params[urlDecodeIfNeeded(key)] = &param
+	}
+	return checkParsingError(parsingErr, key, value)
+}
+
+func checkParsingError(parsingErr error, key string, value interface{}) error {
+	if parsingErr != nil {
+		err := &SnowflakeError{
+			Number:      ErrCodeTomlFileParsingFailed,
+			Message:     errMsgFailedToParseTomlFile,
+			MessageArgs: []interface{}{key, value},
+		}
+		return err
+	}
+	return nil
+}
+
+func parseInt(i interface{}) (int, error) {
+	v, ok := i.(string)
+	if !ok {
+		num, ok := i.(int)
+		if !ok {
+			return 0, errors.New("failed to parse the value to integer")
+		}
+		return num, nil
+	}
+	num, err := strconv.Atoi(v)
+
+	if err != nil {
+		return num, err
```

Comment:
Comparing to `parseBool`, why do we pass the error from `strconv` here instead of making our own? If intended, this `if` is redundant

Reply:
I replaced the err to the same message above. Thank you